addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/ActiveScanJob.java

```diff
@@ -130,57 +128,14 @@ private void parsePolicyDefinition(Object policyDefn, AutomationProgress progres
                     policyDefnData,
                     this.policyDefinition,
                     this.getName(),
-                    new String[] {RULES_ELEMENT_NAME},
+                    new String[] {PolicyDefinition.RULES_ELEMENT_NAME},
                     progress);
 
-            ScanPolicy scanPolicy = new ScanPolicy();
-            PluginFactory pluginFactory = scanPolicy.getPluginFactory();
-
-            Object o = policyDefnData.get(RULES_ELEMENT_NAME);
-            if (o instanceof ArrayList<?>) {
-                ArrayList<?> ruleData = (ArrayList<?>) o;
-                for (Object ruleObj : ruleData) {
-                    if (ruleObj instanceof LinkedHashMap<?, ?>) {
-                        LinkedHashMap<?, ?> ruleMap = (LinkedHashMap<?, ?>) ruleObj;
-                        Integer id = (Integer) ruleMap.get("id");
-                        Plugin plugin = pluginFactory.getPlugin(id);
-                        if (plugin != null) {
-                            AttackStrength strength =
-                                    JobUtils.parseAttackStrength(
-                                            ruleMap.get("strength"), this.getName(), progress);
-                            AlertThreshold threshold =
-                                    JobUtils.parseAlertThreshold(
-                                            ruleMap.get("threshold"), this.getName(), progress);
-
-                            Rule rule = new Rule();
-                            rule.setId(id);
-                            rule.setName(plugin.getName());
-                            if (threshold != null) {
-                                rule.setThreshold(threshold.name().toLowerCase());
-                            }
-                            if (strength != null) {
-                                rule.setStrength(strength.name().toLowerCase());
-                            }
-                            this.getData().getPolicyDefinition().addRule(rule);
-
-                        } else {
-                            progress.warn(
-                                    Constant.messages.getString(
-                                            "automation.error.ascan.rule.unknown",
-                                            this.getName(),
-                                            id));
-                        }
-                    }
-                }
-            } else if (o != null) {
-                progress.warn(
-                        Constant.messages.getString(
-                                "automation.error.options.badlist",
-                                this.getName(),
-                                RULES_ELEMENT_NAME,
-                                o));
-            }
-
+            this.getData()
```

Comment:
I think the whole method should be part of `parsePolicyDefinition`, callers should not need to copy the rules or validate the type, it will end up duplicated in each job (and the rules are really internal to the policy definition).